nx/lib/nx.ex

```diff
@@ -3265,6 +3265,61 @@ defmodule Nx do
     end)
   end
 
+  @doc ~S"""
+  Split tensor or list into train and test subsets.
+
+  ## Examples
+
+    Split into 80% for training and 20% for testing.
+
+    iex> {_train, _test} = Nx.split(Nx.tensor([[3, 6, 5], [26, 75, 3], [23, 4, 1]]), split: 0.8)
+    {#Nx.Tensor<
+      s64[2][3]
+      [
+        [3, 6, 5],
+        [26, 75, 3]
+      ]
+      >, #Nx.Tensor<
+        s64[1][3]
+        [
+          [23, 4, 1]
+        ]
+      >
+    }
+
+    iex> {_train, _test} = Nx.split([[3, 6, 5], [26, 75, 3], [23, 4, 1]], split: 0.8)
+    {[[3, 6, 5], [26, 75, 3]], [[23, 4, 1]]}
+  """
+  @doc type: :shape
+  def split(tensor, opts \\ [])
+
+  def split(tensor, opts) when is_tensor(tensor) do
+    {train, test} =
+      tensor
+      |> to_list()
```

Comment:
For tensors you should be using Nx.slice_along_axis instead of converting to a list. 

Also I don't think we should support lists as inputs here